subprojects/core-api/src/main/java/org/gradle/api/artifacts/ModuleDependencyCapabilitiesHandler.java

```diff
@@ -41,4 +42,30 @@ public interface ModuleDependencyCapabilitiesHandler {
      * @param capabilityNotations the capability {@linkplain ConfigurationPublications#capability(Object) notations} (e.g. group:name:version), {@linkplain Provider Providers} of any notation are also accepted
      */
     void requireCapabilities(Object... capabilityNotations);
+
+    /**
+     * Require a capability of a component based on the name of the feature provided by the component.
+     * <p>
+     * A capability is derived from a feature based on the module identity of the component that a dependency
+     * resolves to. For example, variant of a component with module identity 'group:name:version' that provides
+     * a feature named 'test-fixtures' would have a capability 'group:name-test-fixtures:version'.
```

Comment:
Lets first see what the previous code would look like before this feature. The user would in practice want some "baz" feature of "foo-bar" and some "bar-baz" feature of "foo".

The user would write something along the lines of:
```
dependencies {
    implementation("org:foo-bar:1.0") {
        capabilities {
            requireCapability("org:foo-bar-baz:0") // version is ignored
        }
    }
    implementation("org:foo:1.0") {
        capabilities {
            requireCapability("org:foo-bar-baz:0") // version is ignored
        }
    }
} 
```

So in practice, the capability conflict was already there. This is a side-effect of the way we model features via capabilities. 

Looking at the source of `ProjectDerivedCapability`, which was previously the "definition" of a capability for a feature, we can see the capability name was computed with:
```
    private static String computeCapabilityName(Project project, @Nullable String featureName) {
        if (featureName == null) {
            return project.getName();
        }
        return project.getName() + "-" + TextUtil.camelToKebabCase(featureName);
    }
```

We can see that in this situation, we would have previously generated the capability names that conflict with each other. 

I'm open to idea here, but this seems more of a problem related to modeling features via capabilities rather than a problem with the new API.

Reply:
OK, good to confirm.  This is no change from prior behavior then, and not a concern.

I don't have any ideas for improving this that don't include setting up an online registry of capabilities somewhere. 😄